services/s3management/accesskeys.go

```diff
@@ -0,0 +1,165 @@
+package s3management
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/hashicorp/terraform-plugin-framework-timeouts/resource/timeouts"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	s3management "github.com/ionos-cloud/sdk-go-s3-management"
+)
+
+type AccesskeyResourceModel struct {
+	AccessKey       types.String   `tfsdk:"accesskey"`
+	SecretKey       types.String   `tfsdk:"secretkey"`
+	CanonicalUserId types.String   `tfsdk:"canonical_user_id"`
+	ContractUserId  types.String   `tfsdk:"contract_user_id"`
+	Description     types.String   `tfsdk:"description"`
+	ID              types.String   `tfsdk:"id"`
+	Timeouts        timeouts.Value `tfsdk:"timeouts"`
+}
+
+// AccessKeyDataSourceModel is used to represent an accesskey for a data source.
+type AccessKeyDataSourceModel struct {
+	AccessKey       types.String `tfsdk:"accesskey"`
+	CanonicalUserId types.String `tfsdk:"canonical_user_id"`
+	ContractUserId  types.String `tfsdk:"contract_user_id"`
+	Description     types.String `tfsdk:"description"`
+	ID              types.String `tfsdk:"id"`
+}
+
+func (c *Client) GetAccessKey(ctx context.Context, accessKeyId string) (s3management.AccessKey, *s3management.APIResponse, error) {
+	accessKey, apiResponse, err := c.client.AccesskeysApi.AccesskeysFindById(ctx, accessKeyId).Execute()
+	apiResponse.LogInfo()
+	return accessKey, apiResponse, err
+}
+
+func (c *Client) ListAccessKeys(ctx context.Context) (s3management.AccessKeyList, *s3management.APIResponse, error) {
+	accessKeys, apiResponse, err := c.client.AccesskeysApi.AccesskeysGet(ctx).Execute()
+	apiResponse.LogInfo()
+	return accessKeys, apiResponse, err
+}
+
+func (c *Client) CreateAccessKey(ctx context.Context, accessKey s3management.AccessKeyCreate, timeout time.Duration) (s3management.AccessKey, *s3management.APIResponse, error) {
+	accessKeyResponse, apiResponse, err := c.client.AccesskeysApi.AccesskeysPost(ctx).AccessKeyCreate(accessKey).Execute()
+	apiResponse.LogInfo()
+	if err == nil && accessKeyResponse.Id != nil {
```

Comment:
i think here would be more readable to return early on error.

Reply:
done